src/soc/hw/noc/test/test_noc_buffer.py

```diff
@@ -0,0 +1,343 @@
+"""
+Test the module noc_buffer.
+"""
+
+import random
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import RisingEdge, FallingEdge
+from cocotb.result import TestFailure
+
+# Mirroring expected contents of the FIFO
+_buffer_flits = []
+_buffer_last = []
+
+@cocotb.test()
+def test_noc_buffer(dut):
+    """
+    Test the module noc_buffer.
+    """
+
+    # Read the parameters back from the DUT to set up our model
+    width = dut.FLIT_WIDTH.value.integer
+    depth = dut.DEPTH.value.integer
+    fullpacket = dut.FULLPACKET.value.integer
+    dut._log.info("NoC buffer for a flit width of {} bits and a depth of {}. ".format(width, depth) +
+                  "FULLPACKET: {}".format(fullpacket))
+
+    # Setup clock
+    cocotb.fork(Clock(dut.clk, 10, "ns").start())
+
+    # Reset
+    dut._log.info("Resetting DUT")
+    dut.rst <= 1
+
+    dut.in_flit <= 0
+    dut.in_valid <= 0
+    dut.in_last <= 0
+    dut.out_ready <= 0
+
+    for _ in range(2):
+        yield RisingEdge(dut.clk)
+    dut.rst <= 0
+
+    # Special test case to test FULLPACKET behavior
+    if fullpacket == 1 and width == 8 and depth == 4:
```

Comment:
So essentially the `fullpacket == 1` testcase is hardcoded to a single configuration? That's fine of course, but could you add a comment in the test manifest that only this configuration is supported? Otherwise it looks like one could simply change the manifest to get a wider sweep, which isn't the case.

Apart from that: the current structure is a bit confusing. How about:

```py
@cocotb.coroutine
def _init_dut():
  # copy all setup/reset functionality here

@cocotb.test()
def test_noc_buffer_nofullpacket(dut):
  if dut.FULLPACKET.value.integer:
    return

  yield _init_dut()
  # ...

@cocotb.test()
def test_noc_buffer_fullpacket(dut):
  # check for supported parameters, return if not supported

  yield _init_dut()
  # ...
```

Reply:
But wouldn't that mean I need two individual manifest files?

Reply:
No, having multiple tests in a single test file is fine and doesn't require multiple manifests.